File: Rookie2EngineerBatch8-BE/schemas/report.py
from pydantic import BaseModel

class ReportRead(BaseModel):
    """Report model"""
    category: str
    total: int = 0
    assigned: int = 0
    available: int = 0
    not_available: int = 0
    waiting_for_recycling: int = 0
    recycled: int = 0

File: Rookie2EngineerBatch8-FE/src/types/report.ts
import { ReportSortOption, SortDirection } from "@/types/enums";

export interface Report {
    category: string;
    total: number;
    assigned: number;
    available: number;
    not_available: number;
    waiting_for_recycling: number;
    recycled: number;
}

export interface ReportListQueryParams {
    page?: number | 1;
    size?: number | 20;
    sort_by?: ReportSortOption;
    sort_direction?: SortDirection;
}
